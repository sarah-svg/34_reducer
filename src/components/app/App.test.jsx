  
import React from 'react';
import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  afterEach(() => cleanup());
  
  it('takes a snapshot test of App', async() => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();

  });

  it('checks the current color', async() => {
    render(<App />);
    const current = screen.getByTestId('current');
    fireEvent.change(current, {
      target: {
        value: '#FF0000'
      }
    });
    const display = await screen.findByTestId('display');
        
    expect(display).toHaveStyle({
      backgroundColor: '#FF0000',
    });
  });

  it('checks the current color after hitting the undo button', async() => {
    render(<App />);
    const current = screen.getByTestId('before');
    fireEvent.change(current, {
      target: {
        value: '#00FF00'
      }
    });
    const display = await screen.findByTestId('display');
        
    expect(display).toHaveStyle({
      backgroundColor: '#FF0000',
    });
  });

  it('checks the current color after hitting the redo button', async() => {
    render(<App />);
    const current = screen.getByTestId('after');
    fireEvent.change(current, {
      target: {
        value: '#FF0000'
      }
    });
    const display = await screen.findByTestId('display');
        
    expect(display).toHaveStyle({
      backgroundColor: '#FF0000',
    });
  });
});
