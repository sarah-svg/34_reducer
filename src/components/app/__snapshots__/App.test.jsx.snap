// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component takes a snapshot test of App 1`] = `
<DocumentFragment>
  <button>
    undo
  </button>
  <button>
    redo
  </button>
  <input
    data-testid="current"
    type="color"
    value="#FF0000"
  />
  <div
    data-testid="display"
    style="background-color: rgb(255, 0, 0); width: 10rem; height: 10rem;"
  />
</DocumentFragment>
`;
